runtime/src/iree/hal/drivers/hip/hip_device.c

```diff
@@ -439,166 +230,195 @@ static iree_status_t iree_hal_hip_device_check_params(
-    tracing_device_interface->hip_symbols = symbols;
-
-    status = iree_hal_stream_tracing_context_allocate(
-        (iree_hal_stream_tracing_device_interface_t*)tracing_device_interface,
-        device->identifier, device->params.stream_tracing, &device->block_pool,
-        host_allocator, &device->tracing_context);
+      tracing_device_interface->base.vtable =
+          &iree_hal_hip_tracing_device_interface_vtable_t;
+      tracing_device_interface->device_context = &device->topology.devices[i];
+      tracing_device_interface->host_allocator = host_allocator;
+      tracing_device_interface->hip_symbols = symbols;
+
+      status = IREE_HIP_RESULT_TO_STATUS(
+          symbols, hipCtxPushCurrent(device->topology.devices[i].hip_context));
+      if (IREE_UNLIKELY(!iree_status_is_ok(status))) {
+        iree_hal_device_release((iree_hal_device_t*)device);
+        IREE_TRACE_ZONE_END(z0);
+        return status;
+      }
+      status = iree_hal_stream_tracing_context_allocate(
+          (iree_hal_stream_tracing_device_interface_t*)tracing_device_interface,
+          device->identifier, device->params.stream_tracing,
+          &device->block_pool, host_allocator,
+          &device->topology.devices[i].tracing_context);
+      status = IREE_HIP_RESULT_TO_STATUS(symbols, hipCtxPopCurrent(NULL));
+      if (IREE_UNLIKELY(!iree_status_is_ok(status))) {
+        iree_hal_device_release((iree_hal_device_t*)device);
+        IREE_TRACE_ZONE_END(z0);
+        return status;
+      }
+    }
   }
 
   // Memory pool support is conditional.
   if (iree_status_is_ok(status) && params->async_allocations) {
-    int supports_memory_pools = 0;
-    status = IREE_HIP_RESULT_TO_STATUS(
-        symbols,
-        hipDeviceGetAttribute(&supports_memory_pools,
-                              hipDeviceAttributeMemoryPoolsSupported,
-                              hip_device),
-        "hipDeviceGetAttribute");
-    device->supports_memory_pools = supports_memory_pools != 0;
+    device->supports_memory_pools = true;
+    for (iree_host_size_t i = 0; i < device->topology.count; ++i) {
+      int supports_memory_pools = 0;
+      status = IREE_HIP_RESULT_TO_STATUS(
+          symbols,
+          hipDeviceGetAttribute(&supports_memory_pools,
+                                hipDeviceAttributeMemoryPoolsSupported,
+                                device->topology.devices[i].hip_device),
+          "hipDeviceGetAttribute");
+      device->supports_memory_pools &= (supports_memory_pools != 0);
+    }
   }
 
   // Create memory pools first so that we can share them with the allocator.
   if (iree_status_is_ok(status) && device->supports_memory_pools) {
-    status = iree_hal_hip_memory_pools_initialize(
-        symbols, hip_device, context, &params->memory_pools, host_allocator,
-        &device->memory_pools);
+    device->supports_memory_pools = false;
+    // TODO (awoloszyn): Figure out how to set up memory pools in a device group
```

Comment:
```suggestion
    // TODO(awoloszyn): Figure out how to set up memory pools in a device group
```

`// TODO(githubname|#issue): `